### PR TITLE
Improve support on Circle CI

### PR DESCRIPTION
**Pull request by orta**
I realised that I made the Circle source work differently to the others, and it was the source of my confusion. 

In doing this PR I figured I'd improve our changelog and get Danger involved in making sure we keep it up to date ;)

### CODE REVIEW

**Comment by ashfurrow** [comment]
Nice, using Danger to build Danger :+1:

**Comment by orta** [comment]
:dancers:

File: Dangerfile [hunk 1]
```diff
@@ -17,3 +17,7 @@ end
 if pr_body.length < 5
   fail "Please provide a changelog summary in the Pull Request description @#{pr_author}"
 end
+
+unless files_modified.any? { |a| a.include?("CHANGELOG.md") }
```
Comment (orta) [inline Dangerfile]: @KrauseFx would it feel reasonable to allow something like this using custom objects that act array-ish but provide dictionary-like DSL attribtues?

``` ruby
warn("Please include a CHANGELOG entry") unless file_modified["CHANGELOG.md"]
```
```

Comment (KrauseFx) [inline Dangerfile]: YES! And if it's in a subfolder, do we want `file_modified["docs/something.md"]` or `file_modified["docs"]["something.md"]`

**Comment by KrauseFx** [comment]
Looks good to me, however I've been using the GitHub releases up until now. I know you prefer `Changelog.md`

File: Dangerfile (hunk 1 above)
Comment (orta) [inline Dangerfile]: I'd opt for just dealing with one level of allowing "docs/thingy"  - I'd imagine some special casing  around `file_modified["docs/"]` working if any file inside that dir is changes.

Other options is to support whatever glob thing CocoaPods supports ( I think it uses `Dir.glob` under the hood )

Comment (KrauseFx) [inline Dangerfile]: Let's use `glob`:

```
file_modifieds["docs/**/*.md"].count == 4

file_modified?["docs/**/Changelog.md"]
```

What do you think?

Comment (orta) [inline Dangerfile]: are it `files_modified` for the first one, and I'm sold. :+1:

**Comment by orta** [comment]
I don't believe in github releases, changelogs should be in the project - not a github specific special web page

**Comment by KrauseFx** [comment]
Okay :rocket:

**Pull request merged**